packages/channels/src/interface/system-api.ts

```diff
@@ -0,0 +1,25 @@
+/**********************************************************************
+ * Copyright (C) 2025 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ ***********************************************************************/
+
+import type { OpenDialogOptions, Uri } from '@podman-desktop/api';
+
+export const SystemApi = Symbol.for('SystemApi');
+
+export interface SystemApi {
+  openDialog(options: OpenDialogOptions): Promise<Uri[] | undefined>;
```

Comment:
We cannot get the result directly, because of the timeout constraint.

The pattern would be to "ask for a file" with this method returning `Promise<void>`, and getting the result in another channel.

Something similar has been done in the dashboard extension, for the logs:

https://github.com/podman-desktop/extension-kubernetes-dashboard/blob/3b38d7e9ee0e748291ed5d9f579fe4f4070c30cf/packages/channels/src/channels.ts#L65-L67